htlcswitch/link.go

```diff
@@ -271,6 +272,14 @@ type ChannelLinkConfig struct {
 	// GetAliases is used by the link and switch to fetch the set of
 	// aliases for a given link.
 	GetAliases func(base lnwire.ShortChannelID) []lnwire.ShortChannelID
+
+	// ShutdownMsg is an optional value that is set if, at the time of
+	// the link being started, persisted shutdown info was found for the
+	// channel. This value being set means that we previously sent a
+	// Shutdown message to our peer, and so we should do so again on
+	// re-establish and should not allow anymore HTLC adds on the outgoing
+	// direction of the link.
+	ShutdownMsg fn.Option[lnwire.Shutdown]
```

Comment:
If we go this route, where we're willing to let the link be shutdown aware, we probably ought to have a Shutdown method on the link too and then we can ditch all of the lifecycle hooks and stuff. When I was working on #8167, @Roasbeef was averse to doing this type of thing. I happen to side with you here that shutdown is ultimately part of the link's lifecycle so this should be OK but I do think we need an explicit ACK from @Roasbeef

Reply:
However, the choice to make this a static field does have implications in regards to the coop close v2, since Shutdown may be transmitted multiple times in an attempt to reset protocol state back to the beginning of shutdown negotiations.

Reply:
Ah - I just realised what could happen:

1) we send Shutdown to the peer with delivery address A
2) they persist that delivery address 
3) we re-send shutdown with a new delivery address B
4) remote side swallows & ignores the message since they have already received a shutdown from us which was meant to be identical to the first one so they dont even bother parsing it fully.
5) The 2 nodes start fee negotiation. But this will fail as soon as the remote side gets a signature from us that doesnt make sense given the transaction they have constructed based on delivery address A
6) remote side does force close instead

Reply:
nit: Since this is in the config, I might also suggest we call this something else besides `ShutdownMsg`. Though the comment on the struct row is sufficiently descriptive I wonder if we would benefit from referring to this as "QueuedShutdown", "PendingShutdown", or "PreexistingShutdown"

Reply:
renamed to "PreviouslySentShutdown" 👍